plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -109,7 +116,16 @@ private StorageEngine createStorageEngine(CatalogMetadata catalog) throws URISyn
     ConnectorType connector = catalog.getConnector();
     switch (connector) {
       case PROMETHEUS:
-        storageEngine = null;
+        PrometheusCatalogMetadata prometheusCatalogMetadata = (PrometheusCatalogMetadata) catalog;
+        PrometheusClient
+            prometheusClient =
+            new PrometheusClientImpl(new OkHttpClient(),
+                new URI(prometheusCatalogMetadata.getUri()));
+        PrometheusConfig prometheusConfig = new PrometheusConfig();
+        if (prometheusCatalogMetadata.getDefaultTimeRange() != null) {
+          prometheusConfig.setDefaultTimeRange(prometheusCatalogMetadata.getDefaultTimeRange());
```

Comment:
q, how is `defaultTimeRange` used?

Reply:
> any tests?

WIP PR, will add more code and tests.

Reply:
Probably you can convert it back to draft PR?